lib/middleware.js

```diff
@@ -207,10 +208,26 @@ exports.requireEventLogging = function (appConfig, appType) {
   }));
 };
 
+exports.isPlanDeprecated = function () {
+  return function (req, res, next) {
+    if (checkIfPlanDeprecated(req.body.plan_id)) {
+      logger.error(`Service instance with the requested plan with id : '${req.body.plan_id}' cannot be created as it is deprecated.`);
+      throw new Forbidden(`Service instance with the requested plan cannot be created as it is deprecated.`);
+    }
+    next();
+  };
+};
 function formatContentSecurityPolicy(policy) {
   return _.map(policy, formatContentSecurityPolicyDirective).join(' ');
 }
 
 function formatContentSecurityPolicyDirective(values, key) {
   return `${key} ${values.join(' ')};`;
-}
\ No newline at end of file
+}
+function checkIfPlanDeprecated(plan_id){
+ const plan_state=_.get(catalog.getPlan(plan_id), 'metadata.state','active');
+ if (plan_state == 'deprecated')
```

Comment:
you can directly do 

`return plan_state == 'deprecated';`

Reply:
Can you also define enum in constant.js for possible states of the plan and then use it here instead of hardcoded strings?